internal/provider/users_datasource.go

```diff
@@ -0,0 +1,251 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	cloudservicev1 "go.temporal.io/cloud-sdk/api/cloudservice/v1"
+	identityv1 "go.temporal.io/cloud-sdk/api/identity/v1"
+
+	"github.com/temporalio/terraform-provider-temporalcloud/internal/client"
+	"github.com/temporalio/terraform-provider-temporalcloud/internal/provider/enums"
+	internaltypes "github.com/temporalio/terraform-provider-temporalcloud/internal/types"
+)
+
+type (
+	usersDataSource struct {
+		client *client.Client
+	}
+
+	usersDataModel struct {
+		ID    types.String    `tfsdk:"id"`
+		Users []userDataModel `tfsdk:"users"`
+	}
+
+	userDataModel struct {
```

Comment:
nit: I might move the data model to another file, considering it's being used in user and users datasource